### PR TITLE
[QA v0.40.0] Demo Improvements

### PR DESCRIPTION
**Pull request by andrijapau**
**Context:**

Demo improvements encountered during the v0.40 quality assurance process.

**Description of the Change:**

A few things were adjusted in this PR,
- Deprecated code update.
- Seed `nx.draw` drawings since they are drawn with random vertex positions. This leads to false positives in our `qml-diff-checker` automated script.
- Epoch count to reproduce features from an older version.
- Add note to `qrack.py` about performance being dependent on the hardware used to execute the demo

### CODE REVIEW

**Review by JerryChen97 — APPROVED** [review]

File: demonstrations/tutorial_equivariant_graph_embedding.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def permute(A, permutation):
 np.fill_diagonal(A, np.zeros(len(A))) 
 
 G1 = nx.Graph(A)
-pos1=nx.spring_layout(G1)
+pos1 = nx.spring_layout(G1, seed=1)
```
Comment (lillian542) [inline demonstrations/tutorial_equivariant_graph_embedding.py:118]: If I were a user, with no context regarding our release process,  I would be unsure what this seed is for. I see how this is a quality of life improvement for us, but are we sure about adding something to the demo? Code with no clear purpose can be quite disorienting.

**Review by lillian542 — COMMENTED** [review]
I'm not 100% sold on the seeding thing, but I'm not vehemently opposed either. How much harder did these graphs make the demo diff check?

File: demonstrations/tutorial_equivariant_graph_embedding.py (hunk 1 above)
Comment (andrijapau) [reply demonstrations/tutorial_equivariant_graph_embedding.py:118]: > How much harder did these graphs make the demo diff check?  

I wouldn’t say they made it *harder*, but definitely more time-consuming. Since I haven’t done demo diffs before, I needed to take extra time to verify why the graphs were showing up.  

> If I were a user, with no context regarding our release process, I would be unsure what this seed is for.  

That’s a fair point. Still, I think it’s reasonable to assume some familiarity with coding and RNG seeding from anyone using this demo. Honestly, understanding how and why `nx.spring_layout` is there might be even more confusing. 😅  

I know this isn’t the strongest argument, but using these magic seeds are already a common practice across the repo (it's where I got the idea 😄 ),

```python
# Let's first set the graph:

import networkx as nx

nodes = n_wires = 20
edges = 60
seed = 1967

g = nx.gnm_random_graph(nodes, edges, seed=seed)
positions = nx.spring_layout(g, seed=seed)

nx.draw(g, with_labels=True, pos=positions)
```

File: demonstrations/tutorial_fixed_depth_hamiltonian_simulation_via_cartan_decomposition.py [hunk 2]
```diff
@@ -338,7 +338,7 @@ def loss(theta):
 
 theta0 = jnp.ones(len(k), dtype=float)
 
-thetas, energy, _ = run_opt(loss, theta0, n_epochs=600, lr=0.05)
+thetas, energy, _ = run_opt(loss, theta0, n_epochs=1200, lr=0.05)
```
Comment (Qottmann) [inline demonstrations/tutorial_fixed_depth_hamiltonian_simulation_via_cartan_decomposition.py:341]: Looks good, though seems 800 or 1000 should also do the trick :)

Comment (andrijapau) [reply demonstrations/tutorial_fixed_depth_hamiltonian_simulation_via_cartan_decomposition.py:341]: Indeed, I noticed that. I will switch it too 1000, its a nice number.

Comment (andrijapau) [reply demonstrations/tutorial_fixed_depth_hamiltonian_simulation_via_cartan_decomposition.py]: Done!

File: demonstrations/tutorial_equivariant_graph_embedding.py (hunk 1 above)
Comment (lillian542) [reply demonstrations/tutorial_equivariant_graph_embedding.py:118]: Resolved in offline conversation, we're in favour of seeding for reproducibility 🚀

**Review by lillian542 — APPROVED** [review]

File: demonstrations/ml_classical_shadows.metadata.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
         }
     ],
     "dateOfPublication": "2022-05-02T00:00:00+00:00",
-    "dateOfLastModification": "2024-10-07T00:00:00+00:00",
+    "dateOfLastModification": "2025-01-08T00:00:00+00:00",
```
Comment (andrijapau) [inline demonstrations/ml_classical_shadows.metadata.json:9]:
```suggestion
    "dateOfLastModification": "2025-01-10T00:00:00+00:00",
```

File: demonstrations/tutorial_equivariant_graph_embedding.metadata.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
         }
     ],
     "dateOfPublication": "2023-07-13T00:00:00+00:00",
-    "dateOfLastModification": "2024-10-07T00:00:00+00:00",
+    "dateOfLastModification": "2025-01-08T00:00:00+00:00",
```
Comment (andrijapau) [inline demonstrations/tutorial_equivariant_graph_embedding.metadata.json:9]:
```suggestion
    "dateOfLastModification": "2025-01-10T00:00:00+00:00",
```

File: demonstrations/tutorial_falqon.metadata.json [hunk 5]
```diff
@@ -9,7 +9,7 @@
         }
     ],
     "dateOfPublication": "2021-05-21T00:00:00+00:00",
-    "dateOfLastModification": "2024-11-06T00:00:00+00:00",
+    "dateOfLastModification": "2025-01-08T00:00:00+00:00",
```
Comment (andrijapau) [inline demonstrations/tutorial_falqon.metadata.json:12]:
```suggestion
    "dateOfLastModification": "2025-01-10T00:00:00+00:00",
```

File: demonstrations/tutorial_fixed_depth_hamiltonian_simulation_via_cartan_decomposition.metadata.json [hunk 6]
```diff
@@ -6,7 +6,7 @@
         }
     ],
     "dateOfPublication": "2024-12-19T00:00:00+00:00",
-    "dateOfLastModification": "2024-01-07T09:00:00+00:00",
+    "dateOfLastModification": "2025-01-08T00:00:00+00:00",
```
Comment (andrijapau) [inline demonstrations/tutorial_fixed_depth_hamiltonian_simulation_via_cartan_decomposition.metadata.json:9]:
```suggestion
    "dateOfLastModification": "2025-01-10T00:00:00+00:00",
```

File: demonstrations/tutorial_measurement_optimize.metadata.json [hunk 7]
```diff
@@ -6,7 +6,7 @@
         }
     ],
     "dateOfPublication": "2021-01-18T00:00:00+00:00",
-    "dateOfLastModification": "2024-12-13T00:00:00+00:00",
+    "dateOfLastModification": "2025-01-08T00:00:00+00:00",
```
Comment (andrijapau) [inline demonstrations/tutorial_measurement_optimize.metadata.json:9]:
```suggestion
    "dateOfLastModification": "2025-01-10T00:00:00+00:00",
```

File: demonstrations/tutorial_qaoa_intro.metadata.json [hunk 8]
```diff
@@ -6,7 +6,7 @@
         }
     ],
     "dateOfPublication": "2020-11-18T00:00:00+00:00",
-    "dateOfLastModification": "2024-11-06T00:00:00+00:00",
+    "dateOfLastModification": "2025-01-08T00:00:00+00:00",
```
Comment (andrijapau) [inline demonstrations/tutorial_qaoa_intro.metadata.json:9]:
```suggestion
    "dateOfLastModification": "2025-01-10T00:00:00+00:00",
```

File: demonstrations/tutorial_qgrnn.metadata.json [hunk 9]
```diff
@@ -6,7 +6,7 @@
         }
     ],
     "dateOfPublication": "2020-07-27T00:00:00+00:00",
-    "dateOfLastModification": "2024-11-06T00:00:00+00:00",
+    "dateOfLastModification": "2025-01-08T00:00:00+00:00",
```
Comment (andrijapau) [inline demonstrations/tutorial_qgrnn.metadata.json:9]:
```suggestion
    "dateOfLastModification": "2025-01-10T00:00:00+00:00",
```

File: demonstrations/ibm_pennylane.py [hunk 10]
```diff
@@ -135,13 +146,16 @@
 H = dataset.hamiltonian
 qubits = 4
 
+# Initialize QiskitRuntimeService
 service = QiskitRuntimeService()
-# Gets a 127 qubit device from IBM
-backend = service.least_busy(n_qubits=127, simulator=False, operational=True)
+
+# Use an online simulator available on IBM Cloud
```
Comment (andrijapau) [inline demonstrations/ibm_pennylane.py:152]:
```suggestion
# Use the `ibmq_qasm_simulator` available on IBM Cloud
```

File: demonstrations/ibm_pennylane.py [hunk 11]
```diff
@@ -135,13 +146,16 @@
...
+backend = service.backend("ibmq_qasm_simulator")
 
 try:
-    # Although we only need 4 qubits, our device has 127 qubits, therefore we initialize with wires=127
-    dev = qml.device("qiskit.remote", wires=127, backend=backend)
+    # Although we only need 4 qubits, our device supports a maximum of 31 qubits, therefore we
+    # initialize with wires=31
```
Comment (andrijapau) [inline demonstrations/ibm_pennylane.py:157]:
```suggestion
    # Although we only need 4 qubits, our device supports a maximum of 31 qubits, therefore we initialize with wires=31
```
